File: TinyScript/java/src/main/java/parser/ast/FunctionArgs.java
package parser.ast;

import parser.util.ParseException;
import parser.util.PeekTokenIterator;

public class FunctionArgs extends ASTNode {
    public FunctionArgs() {
        super();
        this.label = "args";
    }

    public static ASTNode parse(PeekTokenIterator it) throws ParseException {

        var args = new FunctionArgs();

        while(it.peek().isType()) {
            var type = it.next();
            var variable = (Variable)Factor.parse(it);
            variable.setTypeLexeme(type);
            args.addChild(variable);

            if(!it.peek().getValue().equals(")")) {
                it.nextMatch(",");
            }
        }

        return args;
    }


}

File: TinyScript/js/src/parser/ast/FunctionArgs.js
const ASTNode = require('./ASTNode')
const ASTNodeTypes = require('./ASTNodeTypes')

class FunctionArgs extends ASTNode {

    constructor(){
        super(ASTNodeTypes.FUNCTION_ARGS, "args")
    }
}

module.exports = FunctionArgs

const {Factor} = require('./index')

FunctionArgs.parse = it => {
    // int a, int b, string c

    const args = new FunctionArgs()

    while(it.peek().isType()) {
        const type = it.next()
        const variable = Factor.parse(it)
        args.addChild(variable)
        variable.setTypeLexeme(type)

        if(it.peek().getValue() !== ')') {
            it.nextMatch(',')
        }
    }
    return args

}